src/pipeline.cc

```diff
@@ -218,8 +218,9 @@ Result Pipeline::AddColorAttachment(Buffer* buf, uint32_t location) {
 Result Pipeline::SetDepthBuffer(Buffer* buf) {
   if (depth_buffer_.buffer != nullptr)
     return Result("can only bind one depth buffer in a PIPELINE");
+  if (buf->GetBufferType() != BufferType::kDepth)
+    return Result("expected a depth buffer");
```

Comment:
Why don't we give more debugging information e.g., buffer name?

Reply:
I believe in both of these cases it should have the correct line number. I'll try to write a test for this later.